### PR TITLE
Monorepo with proxy method and update to 1.48.0

### PR DESCRIPTION
**Pull request by Simon-Laux**
follow up to #144

should be squashed before merging

before merging we should wait until the next official version (that includes the monorepo change) is released

This proxy solution is neither generic nor pretty, but it works.

### CODE REVIEW

File: chat.delta.desktop.yml [hunk 1]
```diff
@@ -112,45 +110,64 @@ modules:
...
-      - env DEBUG="*" npm install --install-links=true --offline --verbose /app/dc-core/deltachat-rpc-server/npm-package
-      - rm -r /app/dc-core
-      - npm run build4production --offline --verbose
-      - npm run pack:generate_config --offline --verbose
+      # patch package.json to exclude the extra architectures that we excluded when building the pnpm store
+      - |
+        jq ".pnpm.supportedArchitectures.os = [\"linux\"] | .pnpm.supportedArchitectures.cpu = [\"x64\",\"arm64\"]" package.json > package.new.json
+        mv package.new.json package.json
+      # install deps from store
       - |
```
Comment (nicodh) [inline chat.delta.desktop.yml:145]: Just for understanding: why is this needed?  Are the dc core binaries not already downloaded in the previous build-commands?

**Review by nicodh — APPROVED** [review]
Reviewed by reading

Docs could be a bit more verbose imho - would shorten the time it needs to understand whats going on

File: record.mjs
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply record.mjs:1]: > How it converts the pnpm format to a npm compatible format

well it doesn't do that. what it does it store the package manifests to the repo and write the list of zip files that flatpak builder should download. that list is called flatpakManifest in this code.
Both package information and the packages that were downloaded by flatpak builder are then replayed via the replay script which also emulates the npm registry.

> since the flatpak builder uses npm

no it just downloads files from a list of files. the repo flatpack-builder-tools has tools to generate those lists from npm, yarn, cargo, ... projects, but not from pnpm projects.

for npm it downloads directly to the npm cache, but the cache for pnpm is much more complex so it is not really possible to replicate this directly for pnpm.

simplified difference:
- npm cache: folder with packages
- pnpm cache: content addressable store that has all files unpacked from the packages (extracts all files, renames them, deduplicates and resorts them)

File: chat.delta.desktop.yml (hunk 1 above)
Comment (Simon-Laux) [reply chat.delta.desktop.yml:145]: it excludes the architectures from installation that we didn't include in the cache, because they are not used.

pnpm downloads all architectures for dc core currently when you do pnpm install.
